test_engine.py

```diff
@@ -0,0 +1,360 @@
+    """
+    Tests the load_data method of the engine.
+
+    Parameters
+    ----------
+        engine: object
+            An instance of the Engine class.
+
+    Returns
+    ----------
+        None
+    """
+
+    get_transforms(engine)
+    engine.load_data()
+
+    assert len(engine.train_dataloader) > 0
+    assert len(engine.test_dataloader) > 0
+    assert isinstance(engine.train_dataloader, MonaiLoader)
+    assert isinstance(engine.val_dataloader, MonaiLoader)
+    assert isinstance(engine.test_dataloader, MonaiLoader)
+
+
+def test_save_checkpoint(engine, ckpt_path="Checkpoint"):
+    """
+    Tests the save_checkpoint method of the engine.
+
+    Parameters
+    ----------
+        engine: object
+            An instance of the Engine class.
+        ckpt_path: str
+            Path of the input directory.
+
+    Returns
+    ----------
+        None
+    """
+
+    assert os.path.exists(ckpt_path)
+
+    loaded_checkpoint = torch.load(ckpt_path)
+
+    engine.load_checkpoint(ckpt_path)
+
+    # verify network state dict match
+    assert set(engine.network.state_dict()) == set(loaded_checkpoint["state_dict"])
+    # verify optimizer state dict match
+    assert set(engine.optimizer.state_dict()) == set(loaded_checkpoint["optimizer"])
+    # verify scheduler state dict match
+    assert set(engine.scheduler.state_dict()) == set(loaded_checkpoint["scheduler"])
+
+
+def test_load_checkpoint(
+    engine, checkpoint_path="Checkpoint", ref_path="ref_Checkpoint"
+):
+    """
+    Tests the load_checkpoint method of the engine.
+
+    Parameters
+    ----------
+        engine: object
+            An instance of the Engine class.
+        checkpoint_path: str
+            Path of the input directory.
+        ref_path: str
+            Path to check the model weights.
+
+    Returns
+    ----------
+        None
+    """
+    get_transforms(engine)
+
+    engine.load_checkpoint(ref_path)
+
+    ref_weights = engine.network.state_dict()
+
+    engine.load_checkpoint(checkpoint_path)
+
+    # Get loaded weights
+    loaded_weights = engine.network.state_dict()
+
+    # Check that weights match
+    for i in ref_weights.keys():
+        assert torch.allclose(ref_weights[i], loaded_weights[i], atol=1e-3)
+
+
+def test_fit(engine, save_path="Checkpoint", ref_path="ref_checkpoint"):
+    """
+    Tests the fit method of the engine.
+
+    Parameters
+    ----------
+
+        engine: object
+            An instance of the Engine class.
+
+        save_path: str
+            Path to save the model weights.
+
+        ref_path: str
+            Path to check the model weights.
+
+    Returns
+    ----------
+        None
+    """
+    get_transforms(engine)
+    engine.load_data()
+
+    # Get initial weights
+    init_weights = [p.clone() for p in engine.network.parameters()]
+
+    # Get loaded weights
+    engine.load_checkpoint(ref_path)
+    ref_weights = engine.network.parameters()
+
+    engine.fit(save_path=save_path)
```

Comment:
how many epochs this fit will run? please set it to 1
Also, the fit function by default doesn't save the weights (not sure why :) ) so please enable save_weights param

Reply:
done